### PR TITLE
feat: Add nonce support

### PR DESCRIPTION
**Pull request by teabolt**
Add ability to specify a nonce that will be added as an attribute to `<script>` tags that are loaded using cors or no-cors mode, i.e. `<script nonce='nonce'>`

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ describe('fetchInject', () => {
   const cssName = 'test.css';
   const scriptText = 'function theBestTeam() { console.log("Dodgers."); }';
   const cssText = '.baseball{ color: blue; }';
+  const nonce = '8IBTHwOdqNKAWeKl7plt8g==';
```
Comment (rufman) [inline test/index.spec.js:10]: can you add a test that tests that no nonce is added if none is given?

File: src/injectors.js [hunk 2]
```diff
@@ -7,10 +7,11 @@ export const head = function(w, d, t, o, r, c, s) {
   s ? s.parentNode.insertBefore(c, s) : d.appendChild(c);
```
Comment (rufman) [inline src/injectors.js:7]: we also want to add the nonce in the non cors case

**Review by rufman — CHANGES_REQUESTED** [review]

File: src/injectors.js [hunk 3]
```diff
@@ -4,13 +4,19 @@ export const head = function(w, d, t, o, r, c, s) {
   (c = d.createElement(t)), (s = d.getElementsByTagName(t)[0]);
   c.appendChild(d.createTextNode(o.text));
   c.onload = r(o);
+  if (o.nonce !== undefined && o.nonce !== null) {
```
Comment (rufman) [inline src/injectors.js:7]: just do `if(!o.nonce)`, nonce as just an empty string is an invalid nonce anyway.

File: src/injectors.js [hunk 4]
```diff
@@ -4,13 +4,19 @@ export const head = function(w, d, t, o, r, c, s) {
...
   s ? s.parentNode.insertBefore(c, s) : d.appendChild(c);
 };
 
-export const headCors = function(w, d, t, u, r, e, c, s) {
+export const headCors = function(w, d, t, u, r, e, c, s, n) {
   (c = d.createElement(t)), (s = d.head.getElementsByTagName(t)[0]);
   c.src = u;
   c.onload = r;
   c.onerror = e;
+  if (n !== undefined && n !== null) {
```
Comment (rufman) [inline src/injectors.js:18]: same here

**Review by rufman — CHANGES_REQUESTED** [review]

**Review by rufman — APPROVED** [review]
